File: backend/app/Models/Employee.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Spatie\Activitylog\LogOptions;
use Spatie\Activitylog\Traits\LogsActivity;

class Employee extends Model
{
  use HasFactory, LogsActivity;

  public function getActivitylogOptions(): LogOptions
  {
    return LogOptions::defaults()
      ->logFillable()
      ->logOnlyDirty()
      ->dontSubmitEmptyLogs();
  }

  protected $fillable = [
    'employee_code',
    'first_name',
    'last_name',
    'email',
    'phone',
    'date_of_birth',
    'gender',
    'department',
    'position',
    'employment_status',
    'employment_type',
    'hire_date',
    'salary',
    'address',
    'avatar_url',
    'contract_end_date',
    'probation_end_date',
    'manager_id',
  ];

  protected $casts = [
    'date_of_birth' => 'date',
    'hire_date' => 'date',
    'salary' => 'decimal:2',
    'contract_end_date' => 'date',
    'probation_end_date' => 'date',
  ];

  protected $appends = ['full_name'];

  public function getFullNameAttribute()
  {
    return "{$this->first_name} {$this->last_name}";
  }

  public function documents()
  {
    return $this->hasMany(EmployeeDocument::class);
  }

  public function attendances()
  {
    return $this->hasMany(Attendance::class);
  }

  public function manager()
  {
    return $this->belongsTo(Employee::class, 'manager_id');
  }

  public function directReports()
  {
    return $this->hasMany(Employee::class, 'manager_id');
  }

  public function user()
  {
    return $this->hasOne(User::class);
  }
}

File: mobile/lib/features/employees/models/employee.dart
class Employee {
  final int id;
  final String employeeCode;
  final String firstName;
  final String lastName;
  final String fullName;
  final String email;
  final String? phone;
  final String? department;
  final String? position;
  final String employmentStatus;
  final String employmentType;
  final String? avatarUrl;

  Employee({
    required this.id,
    required this.employeeCode,
    required this.firstName,
    required this.lastName,
    required this.fullName,
    required this.email,
    this.phone,
    this.department,
    this.position,
    required this.employmentStatus,
    required this.employmentType,
    this.avatarUrl,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json['id'],
      employeeCode: json['employee_code'] ?? '',
      firstName: json['first_name'] ?? '',
      lastName: json['last_name'] ?? '',
      fullName: json['full_name'] ?? '',
      email: json['email'] ?? '',
      phone: json['phone'],
      department: json['department'],
      position: json['position'],
      employmentStatus: json['employment_status'] ?? 'active',
      employmentType: json['employment_type'] ?? 'full_time',
      avatarUrl: json['avatar_url'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'employee_code': employeeCode,
      'first_name': firstName,
      'last_name': lastName,
      'full_name': fullName,
      'email': email,
      'phone': phone,
      'department': department,
      'position': position,
      'employment_status': employmentStatus,
      'employment_type': employmentType,
      'avatar_url': avatarUrl,
    };
  }
}
